css-viewport/Overview.bs

```diff
@@ -526,6 +526,39 @@ The <dfn export>unscaled</dfn> value of a CSS length relative to an element is t
 	The {{Window/devicePixelRatio}} of a frame is multiplied by the [=effective zoom=] inherited by its parent frame.
 </h3>
 
+<h2 id=extensions-to-the-window-interface>Extensions to the {{Window}} Interface</h2>
+
+<pre class=idl>
+partial interface Window {
+[SameObject, Replaceable] readonly attribute Viewport? viewport;
+};
+</pre>
+
+<h2 id=viewport>Viewport</h2>
+
+<h3 id="the-viewport-interface">The {{Viewport}} Interface</h3>
+
+<pre class=idl>
+[Exposed=Window]
+interface Viewport : EventTarget {
+  readonly attribute FrozenArray&lt;DOMRect> segments;
+};
+</pre>
+
+<h2 id='segments'>
+	The <dfn attribute for=Viewport>segments</dfn> property
+</h2>
+
+The segments property is an array of DOMRects that represent the dimensions of each existing viewport segment. Additonal details about the definition of a viewport segment can be found here : [[css-env-1#viewport-segments]].
+
+The {{segments}} property represent an immutable snapshot of the device {{segments}} when the value was queried.
+If the device state changes (rotation, window resize, <a spec="DEVICE-POSTURE">posture</a> change), the value previously retrieved will be invalid. Developers can listen to {{ScreenOrientation/onchange|Screen Orientation}}, {{Window/resize|Window Resize}}, or {{DevicePosture/onchange|Posture}} changes events to detect device state changes.
+
+The {{segments}} attribute must run these steps:
+1. If the {{Viewport}}'s associated {{Document}} is not <a>fully active</a>, return 0.
+2. Returns null if there is only a single viewport segment and abort these steps.
```

Comment:
Why not return one segment?

Reply:
Well it’s useless information since in a foldable device the state in which we have one segment the value would be the viewport width and height. It provides no additional information that the developer already has. When the device has one segment it should be treated as a normal device.

Reply:
Yes but it's more consistent to just return the segments available, even if there is only one. Then developers don't have to put more conditionals in their code.

Reply:
Also on Android when the device is a state where there is no multiple segments the API basically returns a null object signifying the developers there is nothing special going on.

Reply:
Let’s take a step back, in my opinion there is no one segment view, the view is either “segmented” or it’s not. So always two or more.

Reply:
Why though? Why shouldn't segments be considered a generalization of viewports? If developers want to know if there is no folding going on, they can look to see if there is one segment.

Reply:
If they want to know if folding state they should use the Device Posture API not the number of segments.

Reply:
ok...yet another reason not to return null then?

Reply:
Ok. It's good that it's consistent with the environment variables.